### PR TITLE
[RAPTOR-10422] add pylint, move code to try avoid circular deps

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
I'm working on migrating test pipelines to harness.
I decided to add pylint, though for now I made pylint always exit with success even if there are errors.

I still moved some code around to get rid of circular import. There is still one import I can not get rid of. @eric-s-s maybe you could help

### CODE REVIEW

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 1]
```diff
@@ -546,9 +546,7 @@ def __init__(
         """
         self._using_default_type_schema = False
         if not type_schema and use_default_type_schema:
-            from datarobot_drum.drum.common import (
-                read_model_metadata_yaml,
-            )  # local import to prevent cyclic dependency
+            from datarobot_drum.drum.model_metadata import read_model_metadata_yaml
 
             type_schema = read_model_metadata_yaml(
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:551]: @eric-s-s could you recommend how to resolve this circular import?

**Comment by yakov-g** [comment]
jenkins run test_functional_r_environment

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py (hunk 1 above)
Comment (eric-s-s) [reply custom_model_runner/datarobot_drum/drum/typeschema_validation.py:551]: yes.  this won't be easy.  conceptually, `common` should not be importing _from_ other places except for the most basic stuff like `enum` and `exceptions`  so these two import lines are _extremely_ suspicious

```
from datarobot_drum.drum.model_metadata import HyperParameterTrafaret

from datarobot_drum.drum.typeschema_validation import (
    get_type_schema_yaml_validator,
    revalidate_typeschema,
)
```

You should probably keep all of the type_schema validation _only_ in that file.  

notice that all of the funcitonality that uses this is in:

```python 
def read_model_metadata_yaml(code_dir) -> PythonTypingOptional[dict]:
    code_dir = Path(code_dir)
    config_path = code_dir.joinpath(MODEL_CONFIG_FILENAME)
    if config_path.exists():
        with open(config_path) as f:
            try:
                model_config = load(f.read(), MODEL_CONFIG_SCHEMA)  #  THIS IS THE ONLY USE OF THE SCHEMA
                if "typeSchema" in model_config:
                    revalidate_typeschema(model_config["typeSchema"])
                model_config = model_config.data
            except YAMLError as e:
                if "found a blank string" in e.problem:
                    print("The model_metadata.yaml file appears to be empty.")
                else:
                    print(e)
                raise SystemExit(1)
```

## actions

1. `read_model_metadata_yaml` should be in its own place
2. the type schema validator should not be reading metadata, it should _only_ validate existing metadata and should always have that passed in.  
3. none of this should be in common

the main ones are to separate the validation logic from the reader logic.  if the validator only ever deals with already-loaded data, then you won't need to worry about this inside of the validator.  whoever calls the validator should both read the data and then pass it to it's caller and that should take care of your issues.

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Review by rvorobii — APPROVED** [review]

**Comment by akshoop** [comment]
Removing core modeling label since @eric-s-s is already reviewing

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 2]
```diff
@@ -523,38 +523,22 @@ class SchemaValidator:
     actual validation on the respective dataframes.
     """
 
-    _DEFAULT_TYPE_SCHEMA_CODEDIR_PATH = os.path.abspath(
+    DEFAULT_TYPE_SCHEMA_CODEDIR_PATH = os.path.abspath(
```
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:526]: what is this and why is it here?

Comment (eric-s-s) [reply custom_model_runner/datarobot_drum/drum/typeschema_validation.py:526]: as far as I can tell, this is a pointer to a random model-metadata.yaml file.   maybe a default one?  if so, this should be moved to the model_metadata.py file and be encapsultaed in `read_default_model_metadata`  or something like that.  we probably should never be explicitly referencing this file.

File: custom_model_runner/datarobot_drum/drum/drum.py [hunk 3]
```diff
@@ -116,12 +116,13 @@ def __init__(self, runtime):
                     "ensure there are no conflicts with other tasks downstream. Disable strict validation if you do "
                     "not want to use the default type schema."
                 )
-                use_default_type_schema = True
+                type_schema = read_model_metadata_yaml(
```
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/drum.py:119]: ahhhh.  now i understand from comment in type_schema_validation,  please have a specific method for reading the default model_metadata.  if we only ever use this to get the typeschema, then please have the method specifically get the type schema.  this should be in `datarobot_drum.drum.model_metadata`

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/drum.py:119]: yeah, makes sense

File: tests/unit/datarobot_drum/model_metadata/test_model_metadata.py [hunk 4]
```diff
@@ -208,8 +208,10 @@ def yaml_str_to_schema_dict(yaml_str: str) -> dict:
         return schema.data
 
     def test_default_typeschema(self):
-        validator = SchemaValidator(type_schema={}, use_default_type_schema=True)
-        assert validator._using_default_type_schema
+        type_schema = read_model_metadata_yaml(SchemaValidator.DEFAULT_TYPE_SCHEMA_CODEDIR_PATH)[
```
Comment (eric-s-s) [inline tests/unit/datarobot_drum/model_metadata/test_model_metadata.py:211]: what is the purpose of this test?  does it test that the default schema is as expected?

Comment (yakov-g) [reply tests/unit/datarobot_drum/model_metadata/test_model_metadata.py:211]: looks like that

File: custom_model_runner/datarobot_drum/drum/drum.py (hunk 3 above)
Comment (eric-s-s) [reply custom_model_runner/datarobot_drum/drum/drum.py:119]: assuming that this change is incoming, I'm going to approve now.  all else looks good

**Review by eric-s-s — APPROVED** [review]
Looks good.  Assuming that moving the `read_default` functionality gets moved to `model_metadata`, this is good to go.  I'm hitting the approve button now so that I don't hold things up any longer.

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by yakov-g** [comment]
jenkins rerun failed

**Comment by yakov-g** [comment]
jenkins run test_functional_general

**Comment by yakov-g** [comment]
jenkins run pipelineController:all-expected-tasks

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Comment by yakov-g** [comment]
jenkins run test_functional_general test_functional_r_environment

**Comment by datarobot-devinfragit3** [comment]
pipelineController framework is deprecated, please migrate CICD for this repository to Harness

**Pull request merged**